faq.php: some definitions: begin improvements

grahamperrin opened the pull request:
<https://www.freshports.org/faq.php#definitions>

A tentative, fairly major rewrite of much of this section.

grahamperrin committed:
faq.php: some definitions: begin improvements

A tentative, fairly major rewrite of much of this section.

grahamperrin commented:
Draft. Some of this might be contentious. 

For what it's worth, I think that we need to reshape and reframe existing text before attempting to weave in: 

- https://github.com/FreshPorts/freshports/issues/609

grahamperrin commented:
Oops, I forgot to reinsert the part that was originally: 

> After installing, ports are identical to packages. 

It's an important point, however: 

- I wouldn't use the word _identical_ (because you emphasised the value of compile-time options, for a custom-built port to result in a package that will differ from a Project-provided package)

I lean towards mentioning [pkg(8)](https://man.freebsd.org/cgi/man.cgi?query=pkg&sektion=8&manpath=freebsd-ports), however: 

- the longer that section becomes, the less likely that it will suit the "hopelessly lazy" …

dlangille commented on www/faq.php:
```diff
@@ -75,38 +75,52 @@
+	must support concurrent releases of each major version of the operating
+	system – and because build hardware and other resources are not
+	infinite: a single version of the OS must build packages for for
+	multiple releases, for as long as the inferior version will be
```
Something better than inferior. perhaps "older", "previous"?

grahamperrin replied:
```suggestion
	multiple releases, for as long as the older version will be
```

dlangille commented on www/faq.php:
```diff
@@ -75,38 +75,52 @@
+	<p>
+	If you need a package that is not yet available, or not yet compiled by
+	the Project on a required version of the OS, then you can install the
+	port.
```
Install the port, or build from ports?

grahamperrin replied:
[`a86d40d` (#610)](https://github.com/FreshPorts/freshports/pull/610/commits/a86d40da59b7cf0820cd0e2502fca1fa2ed9a170)

dlangille commented on www/faq.php:
```diff
@@ -75,38 +75,52 @@
-
+	If you install a port, the source will be downloaded, patched if 
+	necessary, compiled and installed. If the port has any default
+	configuration option that is preset by the maintainer, you can choose
```
"Some ports have default configuration options. During the build process, you can set these various options."

There's a mix of plural and singular.

grahamperrin replied:
Thanks. I was also not particularly happy with the mixture. I'm vaguely trying to convey: 

- some ports have no more than one option
- wherever there's an option, there's the maintainer-set default (true?)

– as succinctly as possible.

grahamperrin replied:
Reminding myself (an example): 

- <https://www.freshports.org/editors/ghostwriter/#config> _config_ anchor for _Configuration options_
- <https://www.freshports.org/editors/ghostwriter/#options> _options_ anchor for _Options name_.

----

Options … configuration options … build options … compile-time options …

On one hand, I quite like the phrase _build options_.

----

On the other hand, let's compare with something on the poudriere side. 

[poudriere-options(8)](https://man.freebsd.org/cgi/man.cgi?query=poudriere-options&sektion=8&manpath=freebsd-ports)

> **poudriere options** – configure the options for a given port

grahamperrin committed:
www/faq.php: older (not inferior)

Accept Dan's suggestion, https://github.com/FreshPorts/freshports/pull/610#pullrequestreview-2470903357

dlangille commented on www/faq.php:
```diff
@@ -75,38 +75,52 @@
 
 	<h3 id="port">What is a port?</h3>
 	<p>
-	If you install the port, the source will be downloaded, patched if necessary,
```
This section contains a highly simplified overview of ports and packages. We have glossed over many details. More experienced users may feel compelled to point out misleading or incorrect points. Instead, they should submit patches whilst keeping the first sentence in mind.

grahamperrin replied:
I'm happy with a fair amount of glossing-over, because: 

- **Definitions for the hopelessly lazy** is a light-hearted heading
- we plead with the reader to be familiar with the ports-related chapter of the FreeBSD Handbook
- the Foundation's blog post concludes with a link to the same chapter
- the [last part of the chapter](https://docs.freebsd.org/en/books/handbook/ports/#ports-broken) has "Fix it!" as point 3 of 4.

dlangille commented on www/faq.php:
```diff
@@ -75,38 +75,52 @@
-	packages.
-
+	If you install a port, the source will be downloaded, patched if 
+	necessary, compiled and installed. If the port has any default
```
"compiled, and installed"

Oxford commas, all, the, way.

grahamperrin replied:
```suggestion
	necessary, compiled, and installed. If the port has any default
```

grahamperrin committed:
faq.php: build from ports

Resolve Dan's https://github.com/FreshPorts/freshports/pull/610#pullrequestreview-2470903451

grahamperrin committed:
faq.php: Oxford commas

Accept Dan's suggestion, https://github.com/FreshPorts/freshports/pull/610#discussion_r1864252914

grahamperrin committed:
faq.php: &#8211; en dashes

grahamperrin commented:
:-) it's OK, this #610 was probably a draft (not ready for review) with the notion of giving thought first to issue #609 (not a PR).
